comp/api/api/def/component.go

```diff
@@ -41,6 +41,24 @@ type endpointProvider struct {
 	handler http.HandlerFunc
 }
 
+// AuthorizedSet is a type to store the authorized config options for the config API
+type AuthorizedSet map[string]struct{}
+
+// AuthorizedConfigPathsCore is the the set of authorized config keys authorized for the
+// config API.
+var AuthorizedConfigPathsCore = buildAuthorizedSet(
```

Comment:
Why make this public ?

Reply:
I made it public because I was iterating over it in a the extension, but we can probably make it private for now. I feel like we're eventually going to refer to this from elsewhere.